### PR TITLE
gitlab: 16.7.4 -> 16.8.1

### PR DESCRIPTION
**Pull request by yayayayaka**
## Description of changes
- https://gitlab.com/gitlab-org/gitlab/-/blob/v16.8.1-ee/CHANGELOG.md

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/version-management/gitlab/default.nix [hunk 1]
```diff
@@ -19,20 +21,54 @@ let
...
+            rustPlatform.cargoSetupHook
+            rustPlatform.bindgenHook
+          ];
+
+          preInstall = ''
+            export CARGO_HOME="$PWD/../.cargo/"
+          '';
+
+          postInstall = ''
+            cp -v $out/lib/ruby/gems/3.1.0/gems/gitlab-glfm-markdown-0.0.10/lib/glfm_markdown.so $out/lib/ruby/gems/3.1.0/gems/gitlab-glfm-markdown-0.0.10/lib/glfm_markdown
```
Comment (ajs124) [inline pkgs/applications/version-management/gitlab/default.nix:69]:
```suggestion
            mv -v $GEM_HOME/gems/${attrs.gemName}-${attrs.version}/lib/glfm_markdown{.so,}
```
something like that is probably nicer. although I hope we won't have to keep this around for so long.

**Comment by cole-h** [comment]
@ofborg eval

**Comment by yayayayaka** [comment]
NixOS test has finally passed :3

**Comment by yayayayaka** [comment]
Deployed gitlab-ee on x86_64-linux and seems to be doing fine.

**Comment by jslight90** [comment]
16.8.1 was just released.

**Comment by yayayayaka** [comment]
Building the NixOS test. If it succeeds, I will hit merge.

**Comment by yayayayaka** [comment]
Result of `nixpkgs-review pr 280837` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>gitaly</li>
    <li>gitlab</li>
    <li>gitlab-ee</li>
    <li>gitlab-elasticsearch-indexer</li>
    <li>gitlab-pages</li>
    <li>gitlab-shell</li>
    <li>gitlab-workhorse</li>
  </ul>
</details>

**Comment by jslight90** [comment]
I was able to successfully build and install 16.8.1 from b28f41f.  I haven't encountered any issues.

**Comment by ajs124** [comment]
seems like you lost the code that drops the rustc runtime dependency somewhere.

it's basically a copypaste of https://github.com/NixOS/nixpkgs/pull/283982 to `gitlab-glfm-markdown`.

should I open a new PR for that or do you want to?

**Comment by yayayayaka** [comment]
Oh, I'm sorry. Probably dropped while squasing everything together by accident. I have opened #285800 to fix that.

**Comment by ajs124** [comment]
no worries. glad I noticed, though. let's maybe try to better keep track of the closure size together, in the future.

thanks again to @yu-re-ka for noticing this initially.

**Pull request merged**